subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/WatchableHierarchies.java

```diff
@@ -93,6 +110,29 @@ public SnapshotHierarchy removeWatchedHierarchiesOverLimit(SnapshotHierarchy roo
         return result;
     }
 
+    private SnapshotHierarchy removeUnwatchableFileSystems(SnapshotHierarchy root, Invalidator invalidator) {
+        SnapshotHierarchy invalidatedRoot = watchableFileSystemDetector.detectUnsupportedFileSystems()
```

Comment:
I think using a `for` loop and getting rid of the `Combiners` class would be better here.

Reply:
I'd love to establish this pattern of using `reduce()` to operate on `SnaphsotHierarchy`s. Let's keep this around for now and discuss how we can do this better if we can.

Reply:
I would prefer if we did this in a separate PR instead of mixing it in here and have another way of combining snapshot hierarchies.